doc/developer-reference/Preset-and-bundle.md

```diff
@@ -0,0 +1,43 @@
+This page deals with the explanation for 3 classes in the code.
+
+## [`Preset`](../../src/libslic3r/Preset.hpp)
+
+As the name might suggest this class deals with presets for various things. It defines an enum `Type` which basically tells you what kind of data the present contains. Below are a few explained and there corresponding UI elements
+
+#### Note: There is a lot of outdated and legacy code in the code base.
+
+- `TYPE_PRINT`: Refers to a process preset. It's called 'Print' probably due to some legacy code.
+
+<img src="../images/process-preset.png" alt="Example Image" width="320">
+
+- `TYPE_FILAMENT`: As the name suggests this preset is for filaments
+
+<img src="../images/filament-preset.png" alt="Example Image" width="320">
+
+- `TYPE_PRINTER`: Preset for printers.
+
+<img src="../images/printer-preset.png" alt="Example Image" width="320">
+
+There are other preset types but some of them are for SLA. Which is legacy code, since SLA printers are no longer supported. Above 3 are the important types.
+
+## [`PresetBundle`](../../src/libslic3r/PresetBundle.hpp)
+
+This is a bundle containing a few types of `PresetCollection`. One bundle has presets for some printers, filaments and some processes (TYPE_PRINT).
+
+`PresetCollection            prints`\
+`PresetCollection            filaments`\
+`PrinterPresetCollection     printers`
+
+each one of these contains a collection of processes, filaments and printers respectively.\
+
+#### Clarification Needed: It is unclear to me if `printers` contain different nozzle sizes of the same printer model or all different printers models by the same vendor or something else. Feel free to add clarification.
```

Comment:
Each nozzle size is a different preset

Reply:
Yes. I also understood that PresetBundle is just a bundle of whatever presets without regard for compatibility with each other. Compatibility is taken care of later while loading the presets to the UI.

Reply:
Sorry. My previous answer wasn't very clear. The printers preset bundle contains all printer presets. It isn't sorted by nozzle size or vendor.

Reply:
yes. I also realized that while digging through the code earlier today. I have already made that update.

Reply:
> Each nozzle size is a different preset

I want to change this and so many other things if I'm honest. As someone relatively new to 3d printing the way slicers deal with their presets and poor override systems is really frustrating. I assume it's historic but I want one preset where some of the values will change if I just select a different nozzle. 

 I want to be able to say things like if this is PETG then just add some z offset. All my PETG has the same Z_ADJUST custom code.